ndbench-dynamodb-plugins/src/main/java/com/netflix/ndbench/plugin/dynamodb/DynamoDBKeyValue.java

```diff
@@ -0,0 +1,233 @@
+			}
+		}
+	}
+
+	@Override
+	public void init(DataGenerator dataGenerator) throws Exception {
+		logger.info("Initing DynamoDB plugin");
+		client = AmazonDynamoDBClientBuilder.standard().withCredentials(awsCredentialsProvider).build();
+		dynamoDB = new DynamoDB(client);
+
+		/*
+		 * Create a table with a primary hash key named 'name', which holds a string.
+		 * Several properties such as provisioned throughput and atribute names are
+		 * defined in the configuration interface.
+		 */
+
+		logger.debug("Creating table if it does not exist yet");
+		table = dynamoDB.createTable(this.config.getTableName(),
+				Arrays.asList(new KeySchemaElement(config.getAttributeName(), KeyType.HASH)),
+				Arrays.asList(new AttributeDefinition("Id", ScalarAttributeType.N),
+						new AttributeDefinition("value", ScalarAttributeType.S)),
+				new ProvisionedThroughput(this.config.getReadCapacityUnits(), this.config.getWriteCapacityUnits()));
+
+		logger.debug("Waiting until the table is in ACTIVE state");
+		table.waitForActive();
+		
+		DescribeTableRequest describeTableRequest = new DescribeTableRequest()
+				.withTableName(this.config.getTableName());
+		TableDescription tableDescription = client.describeTable(describeTableRequest).getTable();
+		logger.info("Table Description: " + tableDescription);
+
+		logger.info("DynamoDB Plugin initialized");
+	}
+
+	/**
+	 * 
+	 * @param key
+	 * @return the item
+	 * @throws Exception
+	 */
+	@Override
+	public String readSingle(String key) throws Exception {
+		Item item = null;
+		try {
+			GetItemSpec spec = new GetItemSpec().withPrimaryKey("Id", key).withConsistentRead(config.consistentRead());
+			item = table.getItem(spec);
+			if (item == null) {
+				return null;
+			}
+		} catch (AmazonServiceException ase) {
+			amazonServiceException(ase);
+		} catch (AmazonClientException ace) {
+			amazonClientException(ace);
+		}
+		return item.toString();
+	}
+
+	/**
+	 * 
+	 * @param key
+	 * @return A string representation of the output of a PutItemOutcome operation.
+	 * @throws Exception
+	 */
+	@Override
+	public String writeSingle(String key) throws Exception {
+		PutItemOutcome outcome = null;
+		try {
+			Item item = new Item().withPrimaryKey("Id", key).withString("value", this.dataGenerator.getRandomValue());
+			// Write the item to the table
+			outcome = table.putItem(item);
+			if (outcome == null) {
+				return null;
+			}
+
+		} catch (AmazonServiceException ase) {
+			amazonServiceException(ase);
+		} catch (AmazonClientException ace) {
+			amazonClientException(ace);
+		}
+		return outcome.toString();
+	}
+
+	@Override
+	public List<String> readBulk(List<String> keys) throws Exception {
+		return null;
+	}
+
+	@Override
+	public List<String> writeBulk(List<String> keys) throws Exception {
+		return null;
+	}
+
+	@Override
+	public void shutdown() throws Exception {
+		try {
+			logger.info("Issuing DeleteTable request for " + config.getTableName());
```

Comment:
why do we want to delete the table at shutdown? Can it happen that we restart the test at a pre-filled table?

Reply:
This is a good question. NDBench provides a prefill. This is the proper way to make your database hot. The issue with a managed service (table) is that you can configure with high WCUs and RCUs pay a lot of money and then forget it. After you are done with your tests we need to make sure that there is a way to clean the resources. What other options do you propose?